Runtime/HistoryItemBehaviour.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System;
+using System.Collections;
+using TMPro;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace BlockadeLabsSDK
+{
+    public class HistoryItemBehaviour : MonoBehaviour
+    {
+        [SerializeField]
+        private RawImage _thumbnailImage;
+
+        [SerializeField]
+        private TMP_Text _descriptionText;
+
+        [SerializeField]
+        private TMP_Text _timestampText;
+
+        [SerializeField]
+        private Toggle _likeToggle;
+
+        [SerializeField]
+        private Button _removeButton;
+
+        [SerializeField]
+        private Button _downloadButton;
+
+        [SerializeField]
+        private MultiToggle _optionsToggle;
+
+        [SerializeField]
+        private GameObject _modelBadge;
+
+        private ImagineResult _imagineResult;
+
+        private void OnEnable()
+        {
+            _likeToggle.onValueChanged.AddListener(OnLikeToggleValueChanged);
+            _removeButton.onClick.AddListener(OnRemoveButtonClicked);
+            _downloadButton.onClick.AddListener(OnDownloadButtonClicked);
+            _optionsToggle.OnValueChanged.AddListener(OnOptionsToggleValueChanged);
+        }
+
+        private void OnDisable()
+        {
+            _likeToggle.onValueChanged.RemoveListener(OnLikeToggleValueChanged);
+            _removeButton.onClick.RemoveListener(OnRemoveButtonClicked);
+            _downloadButton.onClick.RemoveListener(OnDownloadButtonClicked);
+            _optionsToggle.OnValueChanged.RemoveListener(OnOptionsToggleValueChanged);
+        }
+
+        private async void OnLikeToggleValueChanged(bool value)
+        {
+            if (_imagineResult == null)
+            {
+                _likeToggle.SetIsOnWithoutNotify(false);
+                return;
+            }
+
+            var result = await ApiRequests.ToggleFavorite(_imagineResult.id);
+            _likeToggle.SetIsOnWithoutNotify(result != null && result.request.isMyFavorite);
+        }
+
+        private void OnRemoveButtonClicked()
+        {
+            Debug.Log($"Remove {_imagineResult.id}");
+            // TODO show confirmation dialog
+        }
+
+        private void OnDownloadButtonClicked()
```

Comment:
should probably remove the download button

Reply:
pretty sure we need to download and load it into the main skybox scene.

Reply:
I think in the website the purpose of this button is to download the PNG or whatever. I'm not sure if we need that functionality here, instead just click the whole thumbnail to download/open it in the scene.